test/Channels.Tests.Performance/MemoryPoolBenchmark.cs

```diff
@@ -0,0 +1,69 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Diagnostics.Windows;
+using BenchmarkDotNet.Jobs;
+
+namespace Channels.Tests.Performance
+{
+    [Config(typeof(Config))]
+    public class MemoryPoolBenchmark
+    {
+        public class Config : ManualConfig
+        {
+            public Config()
+            {
+                Add(Job.Default.
+                    With(Platform.X64).
+                    With(Jit.RyuJit).
+                    With(Runtime.Clr).
+                    WithIterationTime(200). // 200ms per iteration
+                    WithWarmupCount(5).
+                    WithTargetCount(10));
+                Add(new MemoryDiagnoser());
+            }
+        }
+
+        public const int InnerLoopCount = 10000;
```

Comment:
Looking at `Leases` count is 4? `OperationsPerInvoke` divides the total time by its value for when you add multiples of the operation in the function.